src/threadsync.cpp

```diff
@@ -368,6 +368,10 @@ bool ThreadSync::wrapperExecutionLockLock()
 {
   int saved_errno = errno;
   bool lockAcquired = false;
+  // Ignore locks if we are about to exit
+  if (DmtcpWorker::exitInProgress()) {
+    return true;
```

Comment:
We should return `false` instead. This will then avoid the need to check for exit status in the unlock code.